homeassistant/components/insteon/api/device.py

```diff
@@ -74,3 +83,70 @@ async def websocket_get_device(
         "aldb_status": str(device.aldb.status),
     }
     connection.send_result(msg[ID], device_info)
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "insteon/device/add",
+        vol.Required(MULTIPLE): bool,
+        vol.Optional(DEVICE_ADDRESS): str,
+    }
+)
+@websocket_api.require_admin
+@websocket_api.async_response
+async def websocket_add_device(
+    hass: HomeAssistant,
+    connection: websocket_api.connection.ActiveConnection,
+    msg: dict,
+) -> None:
+    """Add one or more Insteon devices."""
+    hass.async_create_task(
+        async_add_devices(address=msg.get(DEVICE_ADDRESS), multiple=msg[MULTIPLE])
+    )
+    connection.send_result(msg[ID])
+
+
+@websocket_api.websocket_command({vol.Required(TYPE): "insteon/device/adding"})
+@websocket_api.require_admin
+@websocket_api.async_response
+async def websocket_notify_on_device_added(
+    hass: HomeAssistant,
+    connection: websocket_api.connection.ActiveConnection,
+    msg: dict,
+) -> None:
+    """Tell Insteon a new device was added."""
+
+    @callback
+    def device_added(address: str, action: DeviceAction):
+        """Forward device events to websocket."""
+        if action == DeviceAction.ADDED:
+            forward_data = {"type": "device_added", "address": address}
+        elif action == DeviceAction.COMPLETED:
+            forward_data = {"type": "linking_stopped", "address": ""}
+        else:
+            return
+        connection.send_message(websocket_api.event_message(msg["id"], forward_data))
+
+    @callback
+    def async_cleanup() -> None:
+        """Remove signal listeners."""
+        devices.unsubscribe(device_added)
+        forward_data = {"type": "unsubscribed"}
+        connection.send_message(websocket_api.event_message(msg["id"], forward_data))
```

Comment:
Cleanup is called when the connection is being torn down or the client unsubscribes, you don't need to send a message. Client can notify itself and if connection is down, there is no message to send.